### PR TITLE
JT Form Generate Labels

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This addresses #4515.  It also removes some code that was unnecessary and some code that was dead.  There is still a bug associated with text inputs that submits the form when the user presses enter.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
9.1.1
```

##### ADDITIONAL INFORMATION
generate labels

### CODE REVIEW

File: awx/ui_next/src/api/models/JobTemplates.js [hunk 1]
```diff
@@ -24,8 +23,11 @@ class JobTemplates extends InstanceGroupsMixin(NotificationsMixin(Base)) {
     return this.http.get(`${this.baseUrl}${id}/launch/`);
   }
 
-  associateLabel(id, label) {
-    return this.http.post(`${this.baseUrl}${id}/labels/`, label);
+  associateLabel(id, label, orgId) {
+    return this.http.post(`${this.baseUrl}${id}/labels/`, {
+      name: label.name,
+      organization: orgId,
```
Comment (AlexSCorey) [inline awx/ui_next/src/api/models/JobTemplates.js:29]: this call is used to generate a brand new label and to associate and existing label.

File: awx/ui_next/src/api/models/JobTemplates.js [hunk 2]
```diff
@@ -35,15 +37,10 @@ class JobTemplates extends InstanceGroupsMixin(NotificationsMixin(Base)) {
...
-      name: label.name,
-      organization: orgId,
-    });
-  }
-
   readCredentials(id, params) {
-    return this.http.get(`${this.baseUrl}${id}/credentials/`, { params });
+    return this.http.get(`${this.baseUrl}${id}/credentials/`, {
+      params,
+    });
```
Comment (AlexSCorey) [inline awx/ui_next/src/api/models/JobTemplates.js:43]: This is just a prettier formatting change

File: awx/ui_next/src/api/models/JobTemplates.js [hunk 3]
```diff
@@ -60,7 +57,9 @@ class JobTemplates extends InstanceGroupsMixin(NotificationsMixin(Base)) {
   }
 
   readAccessList(id, params) {
-    return this.http.get(`${this.baseUrl}${id}/access_list/`, { params });
+    return this.http.get(`${this.baseUrl}${id}/access_list/`, {
```
Comment (AlexSCorey) [inline awx/ui_next/src/api/models/JobTemplates.js:60]: This is just a prettier formatting change

File: awx/ui_next/src/screens/Template/shared/LabelSelect.jsx [hunk 4]
```diff
@@ -57,13 +57,26 @@ function LabelSelect({ value, placeholder, onChange, onError }) {
     <Select
       variant={SelectVariant.typeaheadMulti}
       onToggle={toggleExpanded}
-      onSelect={onSelect}
+      onSelect={(e, item) => {
```
Comment (AlexSCorey) [inline awx/ui_next/src/screens/Template/shared/LabelSelect.jsx:60]: When the label gets to the custom hook `useSyncedSelectedValue` it expects the label to be an object.

**Review by jlmitch5 — APPROVED** [review]
looks good, and thanks for working through it with me!

**Pull request merged**